tools/http3-test/src/lib.rs

```diff
@@ -0,0 +1,437 @@
+//! match http3_conn.poll(&mut conn) {
+//!     Ok((stream_id, quiche::h3::Event::Headers(headers))) => {
+//!         test.add_response_headers(stream_id, &headers);
+//!     },
+//!
+//!     Ok((stream_id, quiche::h3::Event::Data)) => {
+//!         let mut buf = [0; 65535];
+//!         if let Ok(read) = http3_conn.recv_body(&mut conn, stream_id, &mut buf)
+//!         {
+//!             test.add_response_body(stream_id, &buf, read);
+//!         }
+//!     },
+//!
+//!     _ => ()
+//! }
+//! # Ok::<(), quiche::h3::Error>(())
+//! ```
+//!
+//! ## Tests assertion
+//!
+//! The [`assert()`] method executes the provided test assertion using the
+//! entire set of [`Http3Req`]s. Calling this prematurely is likely to result
+//! in failure, so it is important to store response data and track the number
+//! of completed requests matches the total for a test.
+//!
+//! ```no_run
+//! # let mut url = url::Url::parse("https://cloudflare-quic.com/b/get").unwrap();
+//! # let mut reqs = Vec::new();
+//!
+//! # let expect_hdrs = Some(vec![quiche::h3::Header::new(":status", "200")]);
+//! # reqs.push(http3test::Http3Req::new("GET", &url, None, expect_hdrs));
+//!
+//! # // Using a closure...
+//! # let assert = |reqs: &[http3test::Http3Req]| {
+//! #   reqs[0].assert_hdrs()
+//! # };
+//!
+//! # let mut test = http3test::Http3Test::new(url, reqs, assert, true);
+//! # let mut config = quiche::Config::new(quiche::PROTOCOL_VERSION).unwrap();
+//! # let scid = [0xba; 16];
+//! # let mut conn = quiche::connect(None, &scid, &mut config).unwrap();
+//! # let h3_config = quiche::h3::Config::new(0, 1024, 0, 0)?;
+//! # let mut http3_conn = quiche::h3::Connection::with_transport(&mut conn, &h3_config)?;
+//! let mut requests_complete = 0;
+//! let request_count = test.requests_count();
+//! match http3_conn.poll(&mut conn) {
+//!     Ok((_stream_id, quiche::h3::Event::Finished)) => {
+//!         requests_complete += 1;
+//!         if requests_complete == request_count {
+//!             test.assert()
+//!         }
+//!     },
+//!     _ => ()
+//! }
+//! # Ok::<(), quiche::h3::Error>(())
+//! ```
+//!
+//! [quiche]: https://github.com/cloudflare/quiche/
+//! [test]: struct.Http3Test.html
+//! [`Http3Test`]: struct.Http3Test.html
+//! [`assert_hdrs()`]: struct.Http3Test.html#method.assert_hdrs
+//! [`requests_count()`]: struct.Http3Test.html#method.requests_count
+//! [`set_concurrency()`]: struct.Http3Test.html#method.set_concurrency
+//! [`add_response_headers()`]: struct.Http3Test.html#method.add_response_headers
+//! [`add_response_body()`]: struct.Http3Test.html#method.add_response_body
+//! [`assert()`]: struct.Http3Test.html#method.assert
+
+#[macro_use]
+extern crate log;
+
+use std::collections::HashMap;
+
+use quiche::h3::Header;
+
+pub const USER_AGENT: &str = "quiche-http3-integration-client";
+
+// Stores the request, the expected response headers, and the actual response.
```

Comment:
Should use `///` instead of `//` for comments to be picked up by rustdoc.

Also, there should be an empty (still commented out though) line between the short description and the long one.

Reply:
Same below.